bricks/test_optimizer/__brick__/test/.test_optimizer.dart

```diff
@@ -10,7 +10,7 @@ import 'package:flutter_test/flutter_test.dart';
 {{/tests}}
 void main() {
 {{#isFlutter}}  goldenFileComparator = _TestOptimizationAwareGoldenFileComparator();{{/isFlutter}}
-{{#tests}}  group('{{#snakeCase}}{{{.}}}{{/snakeCase}}', () { {{#snakeCase}}{{{.}}}{{/snakeCase}}.main(); });
+{{#tests}}  group('{{{.}}}', () { {{#snakeCase}}{{{.}}}{{/snakeCase}}.main(); });
```

Comment:
In the algorithm that build the final report, I would like be able to construct the full file name and keep the initial file name with package. **I use the same string that used from import.**

As I explains in the PR description, my goal is that the report in using or without using optimization is the same. In fact, optimization is the implementation detail that seem to me useful to hide.

I can't find a example of problematic case

Here an example of what generated for counter example.


`dart
// GENERATED CODE - DO NOT MODIFY BY HAND
// Consider adding this file to your .gitignore.

import 'dart:io';

import 'package:flutter_test/flutter_test.dart';


import 'app_test.dart' as app_test_dart;
import 'counter/cubit/counter_cubit_test.dart' as counter_cubit_counter_cubit_test_dart;
import 'counter/view/counter_page_test.dart' as counter_view_counter_page_test_dart;
import 'counter/view/counter_view_test.dart' as counter_view_counter_view_test_dart;

void main() {
  goldenFileComparator = _TestOptimizationAwareGoldenFileComparator();
  group('app_test.dart', () { app_test_dart.main(); });
  group('counter/cubit/counter_cubit_test.dart', () { counter_cubit_counter_cubit_test_dart.main(); });
  group('counter/view/counter_page_test.dart', () { counter_view_counter_page_test_dart.main(); });
  group('counter/view/counter_view_test.dart', () { counter_view_counter_view_test_dart.main(); });
}


class _TestOptimizationAwareGoldenFileComparator extends LocalFileComparator {
  final List<String> goldenFilePaths;

  _TestOptimizationAwareGoldenFileComparator()
      : goldenFilePaths = _goldenFilePaths,
        super(_testFile);

  static Uri get _testFile {
    final basedir =
        (goldenFileComparator as LocalFileComparator).basedir.toString();
    return Uri.parse("$basedir/.test_optimizer.dart");
  }

  static List<String> get _goldenFilePaths =>
      Directory.fromUri((goldenFileComparator as LocalFileComparator).basedir)
          .listSync(recursive: true, followLinks: true)
          .whereType<File>()
          .map((file) => file.path)
          .where((path) => path.endsWith('.png'))
          .toList();

  @override
  Uri getTestUri(Uri key, int? version) {
    final keyString = key.path;
    return Uri.parse(goldenFilePaths
        .singleWhere((goldenFilePath) => goldenFilePath.endsWith(keyString)));
  }
}

`

Reply:
Can we exclude this change from this PR as it's not directly related and is susceptible to issues as @wolfenrain pointed out?

Reply:
@felangel  In my opinion it's related. Otherwise, the final report is less clear.  Below an example of report if I exclude this change. Le file name will be counter_cubit_counter_cubit_test_dart


``` dart


CounterCubit initial state is 0 /Users/abdelaziz_gacemi/Devhome/GitProjects/opensource/bloc/examples/flutter_counter/test/counter_cubit_counter_cubit_test_dart (FAILED)
00:02 +13 -1: Some tests failed.
Failing Tests:
 - test/counter_cubit_counter_cubit_test_dart 
 	- [FAILED] CounterCubit initial state is 0



```

Reply:
@felangel  @wolfenrain I rebased from main branch to get the last commit to deal with special character.. I made some tests with a file name that contains ' like widget'_special_test.dart.  Below, results: 

- with`flutter test` : KO
- with `very_good test` in main branch: KO
- like main branch, on my branch is KO

The generated .test_optimizer.dat look like this:
```dart

import 'widget[1]_special_test.dart' as _a;
import 'widget\'_special_test.dart' as _b;
import 'widget_test.dart' as _c;

void main() {
  goldenFileComparator = _TestOptimizationAwareGoldenFileComparator();
  group('widget[1]_special_test.dart', () { _a.main(); });
  group('widget\'_special_test.dart', () { _b.main(); });
  group('widget_test.dart', () { _c.main(); });
}
```
and the obtained error look like: 
`
test/.test_optimizer.dart:10:8: Error: Error when reading 'test/widget': No such file or directory
import 'widget'_special_test.dart' as _b;
       ^

test/.test_optimizer.dart:16:48: Error: Undefined name '_b'.
  group('widget&#x27;_special_test_dart', () { _b.main(); });
                                               ^^
`

Could we exclude this spcial use case from the scope of this PR as this case is also problematic case for `flutter test` and `very_good test` on main branch

Thank you